src/app/drive/components/ShareDialog/ShareDialog.tsx

```diff
@@ -308,6 +333,51 @@ const ShareDialog = (props: ShareDialogProps): JSX.Element => {
     closeSelectedUserPopover();
   };
 
+  const onPasswordCheckboxChange = useCallback(() => {
+    if (!isPasswordProtected) {
+      setOpenPasswordInput(true);
+    } else {
+      setOpenPasswordDisableDialog(true);
+    }
+  }, [isPasswordProtected]);
+
+  const onSavePublicSharePassword = useCallback(
+    async (passwordValue: string) => {
+      try {
+        let sharingInfo = sharingMeta;
+
+        if (!sharingInfo) {
```

Comment:
If users click on "protect with password" and the item is not being shared yet, we create one automatically for them.